src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs

```diff
@@ -0,0 +1,351 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.Emit;
+using Datadog.Trace.ClrProfiler.Helpers;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.Integrations
+{
+    /// <summary>
+    /// Tracing integration for GraphQL.Server.Transports.AspNetCore
+    /// </summary>
+    public static class GraphQLIntegration
+    {
+        private const string IntegrationName = "GraphQL";
+        private const string ServiceName = "graphql";
+
+        private const string ParseOperationName = "graphql.parse"; // Instrumentation not yet implemented
+        private const string ValidateOperationName = "graphql.validate";
+        private const string ExecuteOperationName = "graphql.execute";
+        private const string ResolveOperationName = "graphql.resolve"; // Instrumentation not yet implemented
+
+        private const string GraphQLAssemblyName = "GraphQL";
+        private const string GraphQLDocumentValidatorInterfaceName = "GraphQL.Validation.IDocumentValidator";
+        private const string GraphQLExecutionResultName = "GraphQL.ExecutionResult";
+        private const string GraphQLExecutionStrategyInterfaceName = "GraphQL.Execution.IExecutionStrategy";
+        private const string GraphQLValidationResultInterfaceName = "GraphQL.Validation.IValidationResult";
+
+        private const string TaskOfGraphQLExecutionResult = "System.Threading.Tasks.Task`1<" + GraphQLExecutionResultName + ">";
+
+        private static readonly ILog Log = LogProvider.GetLogger(typeof(GraphQLIntegration));
+
+        /// <summary>
+        /// Wrap the original method by adding instrumentation code around it.
+        /// </summary>
+        /// <param name="documentValidator">The instance of GraphQL.Validation.IDocumentValidator.</param>
+        /// <param name="originalQuery">The source of the original GraphQL query.</param>
+        /// <param name="schema">The GraphQL schema.</param>
+        /// <param name="document">The GraphQL document.</param>
+        /// <param name="rules">The list of validation rules.</param>
+        /// <param name="userContext">The user context.</param>
+        /// <param name="inputs">The input variables.</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        /// <returns>The original method's return value.</returns>
+        [InterceptMethod(
+            TargetAssembly = GraphQLAssemblyName,
+            TargetType = GraphQLDocumentValidatorInterfaceName,
+            TargetSignatureTypes = new[] { GraphQLValidationResultInterfaceName, ClrNames.Ignore, ClrNames.Ignore, ClrNames.Ignore, ClrNames.Ignore, ClrNames.Ignore, ClrNames.Ignore })]
```

Comment:
Do all the types always differ aside from the return?

Reply:
I can make the TargetSignatureTypes more specific, will fix.

Reply:
Just to post my thinking for posterity:
for a target_signature array: `new [] { "System.Threading.Task", "_", "_"};`
on wrapper: `Task Execute(object instance, object param1, object param2);`
it would match `await instance.Execute(LibraryClass1 param1, LibraryClass2 param2);`
it would also match `await instance.Execute(int param1, string param2);`
The second match would bomb when trying to resolve the method on our integration.

Reply:
Fixed in 2308431a22661e1f16f1b3c63d878e6f1e5c0acd